#! /bin/sh /usr/share/dpatch/dpatch-run
## echo-expansion-fix.dpatch by Joachim Schrod
##
## DP: This is a fix of a bug posted on the Xindy mailinglist.
## DP: http://sourceforge.net/mailarchive/message.php?msg_name=AANLkTimSCmPG47e6oProahOmz3Q5HYuCQ2BqxmruJ51k@mail.gmail.com

@DPATCH@
diff --git a/modules/base/tex.xdy b/modules/base/tex.xdy
index d3c23cb..af006ba 100644
--- a/modules/base/tex.xdy
+++ b/modules/base/tex.xdy
@@ -16,15 +16,10 @@
 ;; keep-blanks is not loaded. Module keep-blanks would set a flag, and
 ;; this code would check for it. We will need a hook in the kernel, to
 ;; achieve that goal.
-;;
-;; FIXME: Regexps with start-anchored matches do not work. They match
-;; everywhere. We think it's better to not handle leading blanks than
-;; to make all blanks disappear, therefore the first rule is not
-;; activated.
 
-;(merge-rule "^ +" "" :eregexp)
+(merge-rule "^ +" "" :eregexp)
 (merge-rule " +$" "" :eregexp)
-(merge-rule " +" " " :eregexp)
+(merge-rule "  +" " " :eregexp)
 
 
 ;;
@@ -55,7 +50,7 @@
 
 (merge-rule "\\[a-zA-Z@]+ *" "" :eregexp)
 (merge-rule "\\[^a-zA-Z@]" "" :eregexp)
-(merge-rule "^[{}$]" "" :eregexp)
+(merge-rule "[{}$]" "" :eregexp)
 
 
 
